### PR TITLE
Add ArchiveSource type

### PR DESCRIPTION
**Pull request by giordano**
Partially address #668 by adding the type `ArchiveSource`

### CODE REVIEW

**Comment by giordano** [comment]
Interestingly enough, the test that is currently failing passes locally for me

**Comment by giordano** [comment]
Ok, should be ready for review now

File: test/building.jl [hunk 1]
```diff
@@ -273,3 +273,39 @@ end
     end
 end
 
+@testset "Building from remote file" begin
```
Comment (giordano) [inline test/building.jl]: I've added a very simple test for building a remote source, as this code path is completely untested.  If you have better ideas about what and how to test here, please share

Comment (staticfloat) [reply test/building.jl]: This seems fine to me; it does mean that our tests may get sad if the network goes down; we could run a local webserver to fix that, but let's cross that bridge once we get to it.

**Comment by staticfloat** [comment]
Wow, this looks really good to me.  Great improvement!  No comments!

**Comment by staticfloat** [comment]
I'm merging, but remember that we'll need to bump the `Manifest` in Yggdrasil before we will see these changes.

**Pull request merged**